CollAction/Dockerfile

```diff
@@ -1,22 +1,32 @@
-﻿FROM microsoft/dotnet:2.1.1-aspnetcore-runtime AS base
-WORKDIR /app
-EXPOSE 80
+﻿# Build webpack bundle 
+# Installs all packages and builds a production version of the javascript and css
+FROM node:10.9-alpine as webpack
+WORKDIR /src
+COPY CollAction/package.json .
+COPY CollAction/package-lock.json .
+RUN npm install
+COPY CollAction/Frontend ./Frontend
+COPY CollAction/webpack.config.js .
+COPY CollAction/tsconfig.json .
+RUN NODE_ENV=production npm run build
 
-FROM microsoft/dotnet:2.1-sdk AS build
+# Build application
+# Installs all packages, compiles the dotnet code and copies the frontend bundle in it
+FROM microsoft/dotnet:2.1-sdk-alpine AS build
 WORKDIR /src
 COPY CollAction/CollAction.csproj CollAction/
-RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
-RUN apt-get update && apt-get install -y nodejs build-essential
 RUN dotnet restore CollAction/CollAction.csproj
 COPY . .
 WORKDIR /src/CollAction
-RUN dotnet build CollAction.csproj -c Release -o /app
+RUN dotnet build CollAction.csproj --no-restore -c Release -o /app
+RUN dotnet publish CollAction.csproj --no-build -c Release -o /app
```

Comment:
I don't think travis caches the docker images in any way (fresh vm each time), so why not just do this:

RUN NODE_ENV=production dotnet build CollAction.csproj -c Release -o /app

This seems like a lot of code for a small speedup that probably doesn't really matter. 

Using the alphine image is a good idea, but I'm not really sure that the rest really matter.